src/mdesktopentry.cpp

```diff
@@ -243,32 +288,51 @@ QString MDesktopEntry::fileName() const
 
 bool MDesktopEntry::contains(const QString &key) const
 {
+    if (key.startsWith(DesktopEntrySection + "/")) {
+        return d_ptr->desktopEntriesMap.contains(key.mid(DesktopEntrySection.length() + 1));
+    }
     return d_ptr->desktopEntriesMap.contains(key);
```

Comment:
If only Desktop Entry content in read into the map, this and similar should fail?